includes/class-blocks.php

```diff
@@ -339,7 +339,19 @@ public static function render_reply_block( $attrs ) {
 		$html = '';
 
 		if ( ! empty( $attrs['url'] ) ) {
-			$html = sprintf(
+			wp_oembed_add_provider( '#https?://mastodon\.social/(@.+)/([0-9]+)#i', 'https://mastodon.social/api/oembed', true );
```

Comment:
isn't this too specific for mastodon.social? what about using the ActivityPub JSON instead?

Reply:
yes, Mastodon.social is the biggest, but I would vote for a generic solution!

Reply:
> what about using the ActivityPub JSON instead?

Oh, that's not something I had considered when I created the issue, but pulling data from the ActivityPub representation instead of trying to embed may be even cleaner, I like the idea!

Reply:
First pass in [3db22d2](https://github.com/Automattic/wordpress-activitypub/pull/1100/commits/3db22d2c9a46e02cef23685fd9c0d50c2a284cbd).
This could use some heavy testing with posts from various different platforms.

Should we try oembed first?